### ISSUE TITLE
Autocomplete component suggestion feature not working with Reactive Forms

### ISSUE THREAD

**Issue by Zaixu** [telerik/kendo-angular#2375]
**Describe the bug**
As soon as you couple the autocomplete component up to reactive forms the value is not updated (until choosing one or blur), thus when you set new filtered data it works on old value and will never come with a proper suggestion.

You will only see the suggestion from the initial selection focus.

**To Reproduce**

Have a look here: [Link to stackblitz](https://stackblitz.com/edit/angular-2gprsl)

**Additional context**

I looked through source and it seems like in the set data function it looks at this.value which is not updated till blur/set value/choose item. I would guess its better with 'this.text' which is the component that the user is writing to. As the user has not chosen/blurred yet.

Else the component would have to update this.value with the this.text

```
    set data(data: any) {
        this._data = data || [];
        this.dataService.data = this._data;

        this.setState(this.value);

        if (this.filterable) {
            this.selectionService.focused = this.isOpen && this.data.length && this.highlightFirst ? this.firstFocusableIndex(0) : -1;
        }

        const shouldSuggest: boolean = this.suggest && this.data.length && this.value; // << this one
        if (shouldSuggest) {
            this.suggestedText = getter(this.dataService.itemAt(0), this.valueField);
        }
    }
```

**Comment by dimitar-pechev** [comment]
@Zaixu, thank you for drawing our attention to this issue. 

Looking at the code, it seems it is actually related to filtering and not to reactive forms. With the `filterable` prop disabled, the `suggest` feature works as expected with reactive forms. Refer to [this example](https://stackblitz.com/edit/angular-2gprsl-edehq2). When we enable filtering, the suggestions are partially broken even without using forms at all. 

I must admin the check for a present `value` in the `data` setter doesn't really make a lot of sense at least at first sight. We'll investigate the problem and provide a fix.

**Comment by ipeshev** [comment]
Fixed in [latest dev version](https://www.telerik.com/kendo-angular-ui-develop/components/installation/development-builds/).

**Comment by dimitar-pechev** [comment]
The fix is shipped with **v4.1.0** of the `@progress/kendo-angular-dropdowns` package.
